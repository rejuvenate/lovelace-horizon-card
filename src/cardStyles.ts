import { css } from 'lit'

export default css`
  :host {
    --hc-primary: var(--primary-text-color);
    --hc-secondary: var(--secondary-text-color);

    --hc-field-name-color: var(--hc-secondary);
    --hc-field-value-color: var(--hc-primary);

    --hc-day-color: #8ebeeb;
    --hc-night-color: #393b78;

    --hc-accent: #d7d7d7;
    --hc-lines: var(--hc-accent);

    --hc-sun-hue: 44;
    --hc-sun-saturation: 93%;
    --hc-sun-lightness: 67%;
    --hc-sun-hue-reduce: 0;
    --hc-sun-saturation-reduce: 0%;
    --hc-sun-lightness-reduce: 0%;
    --hc-sun-color: hsl(
      calc(var(--hc-sun-hue) - var(--hc-sun-hue-reduce)),
      calc(var(--hc-sun-saturation) - var(--hc-sun-saturation-reduce)),
      calc(var(--hc-sun-lightness) - var(--hc-sun-lightness-reduce))
    );

    --hc-moon-hue: 52;
    --hc-moon-saturation: 77%;
    --hc-moon-lightness: 57%;
    --hc-moon-saturation-reduce: 0%;
    --hc-moon-lightness-reduce: 0%;
    --hc-moon-color: hsl(
      var(--hc-moon-hue),
      calc(var(--hc-moon-saturation) - var(--hc-moon-saturation-reduce)),
      calc(var(--hc-moon-lightness) - var(--hc-moon-lightness-reduce))
    );
    --hc-moon-shadow-color: #eeeeee;
    --hc-moon-spot-color: rgba(170, 170, 170, 0.1);
  }

  :host(.horizon-card-dark) {
    --hc-accent: #464646;
    --hc-moon-saturation: 80%;
    --hc-moon-lightness: 74%;
    --hc-moon-shadow-color: #272727;
  }

  .horizon-card {
    padding: 0.5em;
    font-family: var(--primary-font-family);
  }

  .horizon-card-field-row {
    display: flex;
    justify-content: space-around;
    margin-top: 1em;
    margin-bottom: -0.3em;
  }

  .horizon-card-text-container {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .horizon-card-field-name {
    color: var(--hc-field-name-color);
  }

  .horizon-card-field-value {
    color: var(--hc-field-value-color);
    font-size: 1.2em;
    line-height: 1.1em;
    text-align: center;
  }

  .horizon-card-field-value-moon-phase {
    font-size: inherit;
  }

  .horizon-card-field-moon-phase {
    --mdc-icon-size: 2em;
    color: var(--primary-color);
  }

  .horizon-card-field-value-secondary {
    font-size: 0.7em;
  }

  .horizon-card-sun-value:before {
    content: "☉";
    padding-right: 0.5em;
  }

  .horizon-card-moon-value:before {
    content: "☽";
    padding-right: 0.5em;
  }

  .horizon-card-header {
    display: flex;
    justify-content: space-around;
    margin-top: 1em;
    margin-bottom: -0.3em;
  }

  .horizon-card-header .horizon-card-text-container {
    font-size: 1.2em;
  }

  .horizon-card-footer {
    margin-bottom: 1em;
  }

  .horizon-card-title {
    margin: 1em 1em 1em 1em;
    font-size: 1.5em;
    color: var(--hc-primary);
  }

  .horizon-card-graph {
    margin: 1em 0.5em 1em 0.5em;
  }

  .horizon-card-graph .dawn {
    fill: var(--hc-night-color);
    stroke: var(--hc-night-color);
  }

  .horizon-card-graph .day {
    fill: var(--hc-day-color);
    stroke: var(--hc-day-color);
  }
`
