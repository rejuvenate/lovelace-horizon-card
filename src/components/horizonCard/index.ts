export * from './HorizonCard'
export * from './HorizonCardContent'
export * from './HorizonCardFooter'
export * from './HorizonCardGraph'
export * from './HorizonCardHeader'
