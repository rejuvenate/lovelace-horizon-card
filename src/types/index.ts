import { LovelaceCardConfig, NumberFormat, TimeFormat } from 'custom-card-helpers'

export type THorizonCardFields = {
  sunrise?: boolean
  sunset?: boolean

  dawn?: boolean
  noon?: boolean
  dusk?: boolean

  azimuth?: boolean
  sun_azimuth?: boolean
  moon_azimuth?: boolean
  elevation?: boolean
  sun_elevation?: boolean
  moon_elevation?: boolean

  moonrise?: boolean,
  moonset?: boolean,
  moon_phase?: boolean
}

export interface IHorizonCardConfig extends LovelaceCardConfig {
  dark_mode?: boolean
  language?: string

  title?: string,
  moon?: boolean,
  time_format?: TimeFormat
  number_format?: NumberFormat

  latitude?: number
  longitude?: number
  elevation?: number
  southern_flip?: boolean
  moon_phase_rotation?: number
  now?: Date
  time_zone?: string
  refresh_period?: number
  debug_level?: number

  fields?: THorizonCardFields
}

export type TSunTimes = {
  now: Date
  dawn?: Date
  dusk?: Date
  midnight: Date
  noon: Date
  sunrise?: Date
  sunset?: Date
}

export type TSunData = {
  readonly azimuth: number
  readonly elevation: number
  readonly times: TSunTimes
  readonly hueReduce: number
  readonly saturationReduce: number
  readonly lightnessReduce: number
}

export type TSunPosition = {
  readonly x: number
  readonly y: number
  readonly horizonY: number
  readonly sunriseX: number
  readonly sunsetX: number
  readonly scaleY: number
  readonly offsetY: number
}

export type TMoonTimes = {
  readonly now: Date,
  readonly moonrise: Date,
  readonly moonset: Date
}

export type TMoonPhase = {
  state: string
  icon: string
}

export type SunCalcMoonPhase =
  'newMoon'
  | 'waxingCrescentMoon'
  | 'firstQuarterMoon'
  | 'waxingGibbousMoon'
  | 'fullMoon'
  | 'waningGibbousMoon'
  | 'thirdQuarterMoon'
  | 'waningCrescentMoon'

export type TMoonData = {
  readonly azimuth: number
  readonly elevation: number
  readonly fraction: number
  readonly phase: TMoonPhase
  readonly phaseRotation: number,
  readonly zenithAngle: number
  readonly parallacticAngle: number
  readonly times: TMoonTimes,
  readonly saturationReduce: number,
  readonly lightnessReduce: number
}

export type TMoonPosition = {
  readonly x: number
  readonly y: number
}

export enum EHorizonCardErrors {
}

export type THorizonCardData = {
  readonly partial: boolean
  readonly latitude: number
  readonly longitude: number
  readonly sunPosition: TSunPosition
  readonly sunData: TSunData
  readonly moonPosition: TMoonPosition
  readonly moonData: TMoonData
}

export enum EHorizonCardI18NKeys {
  Azimuth = 'azimuth',
  Dawn = 'dawn',
  Dusk = 'dusk',
  Elevation = 'elevation',
  Noon = 'noon',
  Sunrise = 'sunrise',
  Sunset = 'sunset',
  Moonrise = 'moonrise',
  Moonset = 'moonset'
}

export type THorizonCardI18NErrorKeys = {
  [key in EHorizonCardErrors]: string
}

export type THorizonCardI18NKeys = { [key in EHorizonCardI18NKeys ]?: string } | { errors: THorizonCardI18NErrorKeys }
