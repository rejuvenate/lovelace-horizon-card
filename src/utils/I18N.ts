import { formatNumber, FrontendLocaleData, LocalizeFunc, NumberFormat, TimeFormat } from 'custom-card-helpers'

import { Constants } from '../constants'
import { THorizonCardI18NKeys } from '../types'

export class I18N {
  private readonly localization: THorizonCardI18NKeys
  private readonly dateFormatter: Intl.DateTimeFormat
  private readonly locale: FrontendLocaleData
  private readonly localizeFunc: LocalizeFunc

  constructor (language: string, timeZone: string, timeFormat: TimeFormat, numberFormat: NumberFormat,
               localizeFunc: LocalizeFunc) {
    this.localization = I18N.matchLanguageToLocalization(language)

    this.dateFormatter = I18N.createDateFormatter(language, timeZone, timeFormat)

    this.locale = {
      language,
      time_format: timeFormat,
      number_format: numberFormat
    }

    this.localizeFunc = localizeFunc
  }

  public formatDateAsTime (date: Date): string {
    let time = this.dateFormatter.format(date)
    if (this.locale.language === 'bg') {
      // Strips " ч." from times in Bulgarian - some major browsers insist on putting it there:
      // https://unicode-org.atlassian.net/browse/CLDR-11545
      // https://unicode-org.atlassian.net/browse/CLDR-15802
      time = time.replace(' ч.', '')
    }
    return time
  }

  public formatDecimal (decimal: number): string {
    return formatNumber(decimal, this.locale)
  }

  /**
   * TR -> TRanslation
   * @param translationKey The key to lookup a translation for
   * @returns The string specified in the translation files
   */
  public tr (translationKey: string): string {
    // if the translation isn't completed in the target language, fall back to english
    // give ugly string for developers who misstype
    return this.localization[translationKey] ?? Constants.FALLBACK_LOCALIZATION[translationKey]
      ?? `Translation key '${translationKey}' doesn't have a valid translation`
  }

  public localize (key: string): string {
    return this.localizeFunc(key)
  }

  private static matchLanguageToLocalization (language: string) {
    let data = Constants.LOCALIZATION_LANGUAGES[language]
    if (data === undefined) {
      // Matches things like en-GB to en, es-419 to es, etc.
      data = Constants.LOCALIZATION_LANGUAGES[language.split('-', 2)[0]]
    }
    if (data === undefined) {
      data = Constants.FALLBACK_LOCALIZATION
    }

    return data
  }

  private static createDateFormatter (language: string, timeZone: string, timeFormat: TimeFormat): Intl.DateTimeFormat {
    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat
    const dateTimeFormatOptions: Intl.DateTimeFormatOptions = {
      hour: 'numeric',
      minute: '2-digit',
      timeZone: timeZone
    }

    // mimics home assistant's logic
    if (timeFormat === 'language' || timeFormat === 'system') {
      const testLanguage = timeFormat === 'language' ? language : undefined
      const test = new Date().toLocaleString(testLanguage)
      dateTimeFormatOptions.hour12 = test.includes('AM') || test.includes('PM')
    } else {
      // Casting to string allows both "time_format: 12" and "time_format: '12'" in YAML
      dateTimeFormatOptions.hour12 = String(timeFormat) === '12'
    }

    let timeLocale = language
    if (!dateTimeFormatOptions.hour12) {
      // Prevents times like 24:00, 24:15, etc. with the 24h clock in some locales.
      // Home Assistant does this only for 'en' but zh-Hant for example suffers from the same problem.
      timeLocale += '-u-hc-h23'
    }
    return new Intl.DateTimeFormat(timeLocale, dateTimeFormatOptions)
  }
}
