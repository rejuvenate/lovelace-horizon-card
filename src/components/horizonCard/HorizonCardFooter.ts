import { html, nothing, TemplateResult } from 'lit'

import {
  EHorizonCardI18NKeys,
  IHorizonCardConfig,
  THorizonCardData,
  THorizonCardFields,
  TMoonTimes,
  TSunTimes
} from '../../types'
import { HelperFunctions } from '../../utils/HelperFunctions'
import { I18N } from '../../utils/I18N'

export class HorizonCardFooter {
  private readonly data: THorizonCardData
  private readonly i18n: I18N
  private readonly sunTimes: TSunTimes
  private readonly moonTimes: TMoonTimes
  private readonly fields: THorizonCardFields
  private readonly azimuths
  private readonly azimuthExtraClasses: string[]
  private readonly elevations
  private readonly elevationExtraClasses: string[]

  constructor (config: IHorizonCardConfig, data: THorizonCardData, i18n: I18N) {
    this.data = data

    this.i18n = i18n
    this.sunTimes = data.sunData.times
    this.moonTimes = data.moonData.times
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    this.fields = config.fields!

    this.azimuths = []
    if (this.fields.sun_azimuth) {
      this.azimuths.push(this.data.sunData.azimuth)
    }
    if (this.fields.moon_azimuth) {
      this.azimuths.push(this.data.moonData.azimuth)
    }
    if (this.fields.sun_azimuth && this.fields.moon_azimuth) {
      this.azimuthExtraClasses = ['horizon-card-sun-value', 'horizon-card-moon-value']
    } else {
      this.azimuthExtraClasses = []
    }

    this.elevations = []
    if (this.fields.sun_elevation) {
      this.elevations.push(this.data.sunData.elevation)
    }
    if (this.fields.moon_elevation) {
      this.elevations.push(this.data.moonData.elevation)
    }
    if (this.fields.sun_elevation && this.fields.moon_elevation) {
      this.elevationExtraClasses = ['horizon-card-sun-value', 'horizon-card-moon-value']
    } else {
      this.elevationExtraClasses = []
    }
  }

  public render (): TemplateResult {
    return html`
      <div class="horizon-card-footer">
        ${
          this.renderRow(
            this.fields.dawn
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Dawn, this.sunTimes.dawn)
              : nothing,
            this.fields.noon
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Noon, this.sunTimes.noon)
              : nothing,
            this.fields.dusk
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Dusk, this.sunTimes.dusk)
              : nothing
          )
        }
        ${
          this.renderRow(
            this.fields.sun_azimuth || this.fields.moon_azimuth
              ? HelperFunctions.renderFieldElements(this.i18n, EHorizonCardI18NKeys.Azimuth, this.azimuths,
                this.azimuthExtraClasses)
              : nothing,
            this.fields.sun_elevation || this.fields.moon_elevation
              ? HelperFunctions.renderFieldElements(this.i18n, EHorizonCardI18NKeys.Elevation, this.elevations,
                this.elevationExtraClasses)
              : nothing
          )
        }
        ${
          this.renderRow(
            this.fields.moonrise
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Moonrise, this.moonTimes.moonrise)
              : nothing,
            this.fields.moon_phase
              ? HelperFunctions.renderMoonElement(this.i18n, this.data.moonData.phase, this.data.moonData.phaseRotation)
              : nothing,
            this.fields.moonset
              ? HelperFunctions.renderFieldElement(this.i18n, EHorizonCardI18NKeys.Moonset, this.moonTimes.moonset)
              : nothing
          )
        }
      </div>
    `
  }

  private renderRow (...args: (typeof nothing | TemplateResult)[]) {
    const nonEmpty = args.filter((tr) => tr !== nothing)
    return nonEmpty.length > 0
      ? html`
        <div class="horizon-card-field-row">
          ${nonEmpty}
        </div>`
      : nothing
  }
}
