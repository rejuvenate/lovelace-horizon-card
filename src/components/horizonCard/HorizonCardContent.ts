import { html, TemplateResult } from 'lit'

import { IHorizonCardConfig,THorizonCardData } from '../../types'
import { I18N } from '../../utils/I18N'
import { HorizonCardFooter } from './HorizonCardFooter'
import { HorizonCardGraph } from './HorizonCardGraph'
import { HorizonCardHeader } from './HorizonCardHeader'

export class HorizonCardContent {
  private config: IHorizonCardConfig
  private data: THorizonCardData
  private i18n: I18N

  constructor (config: IHorizonCardConfig, data: THorizonCardData, i18n: I18N) {
    this.config = config
    this.data = data
    this.i18n = i18n
  }

  render (): TemplateResult {
    return html`
      <ha-card>
        <div class="horizon-card">
          ${ this.renderHeader() }
          ${ this.renderGraph() }
          ${ this.renderFooter() }
        </div>
      </ha-card>
    `
  }

  private renderHeader (): TemplateResult {
    return new HorizonCardHeader(this.config, this.data, this.i18n).render()
  }

  private renderGraph (): TemplateResult {
    return new HorizonCardGraph(this.config, this.data).render()
  }

  private renderFooter (): TemplateResult {
    return new HorizonCardFooter(this.config, this.data, this.i18n).render()
  }
}
