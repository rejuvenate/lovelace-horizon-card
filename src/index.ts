import './components/horizonCard'
import './components/horizonCardEditor'

