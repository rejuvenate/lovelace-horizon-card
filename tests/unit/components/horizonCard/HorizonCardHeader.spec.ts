import { NumberFormat, TimeFormat } from 'custom-card-helpers'

import { HorizonCardHeader } from '../../../../src/components/horizonCard'
import { IHorizonCardConfig, THorizonCardData, TSunData, TSunTimes } from '../../../../src/types'
import { I18N } from '../../../../src/utils/I18N'
import { TemplateResultTestHelper } from '../../../helpers/TestHelpers'

jest.mock('../../../../src/utils/I18N', () => require('../../../mocks/I18N'))

describe('HorizonCardHeader', () => {
  const i18n = new I18N('en', 'UTC', TimeFormat.language, NumberFormat.language, (key) => key)

  describe('render', () => {
    it('renders the title if it is present in the configuration', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        title: 'test',
        fields: {}
      }

      const data = {
        sunData: {}
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })

    it('does not render the title if it is not present in the configuration', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        fields: {}
      }

      const data = {
        sunData: {}
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })

    it('renders the sunrise field when it is present in the data and it is activated on the config', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        fields: {
          sunrise: true
        }
      }

      const data = {
        sunData: {
          times: {
            now: new Date(0),
            noon: new Date(0),
            midnight: new Date(0),
            sunrise: new Date(0)
          } as TSunTimes
        } as TSunData
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })

    it('does not render the sunrise field when it is present in the data but it is disabled on the config', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        fields: {
          sunrise: false
        }
      }

      const data = {
        sunData: {
          times: {
            now: new Date(0),
            noon: new Date(0),
            midnight: new Date(0),
            sunrise: new Date(0)
          } as TSunTimes
        } as TSunData
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })

    it('does not render the sunrise field when it is not present in the data but it is activated on the config', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        fields: {
          sunrise: true
        }
      }

      const data = {
        sunData: {
          times: {}
        }
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })

    it('does not render the sunrise field when it is not present in the data and it is disabled on the config', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        fields: {
          sunrise: false
        }
      }

      const data = {
        sunData: {
          times: {}
        }
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })

    it('renders the sunset field when it is present in the data and it is activated on the config', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        fields: {
          sunset: true
        }
      }

      const data = {
        sunData: {
          times: {
            sunset: new Date(0)
          } as TSunTimes
        } as TSunData
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })

    it('does not render the sunset field when it is present in the data but it is disabled on the config', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        fields: {
          sunset: false
        }
      }

      const data = {
        sunData: {
          times: {
            now: new Date(0),
            noon: new Date(0),
            midnight: new Date(0),
            sunrise: new Date(0)
          } as TSunTimes
        } as TSunData
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })

    it('does not render the sunset field when it is not present in the data but it is activated on the config', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        fields: {
          sunset: true
        }
      }

      const data = {
        sunData: {
          times: {}
        }
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })

    it('does not render the sunset field when it is not present in the data and it is disabled on the config', async () => {
      const config: IHorizonCardConfig = {
        type: 'horizon-card',
        fields: {
          sunset: false
        }
      }

      const data = {
        sunData: {
          times: {}
        }
      } as THorizonCardData

      const horizonCardHeader = new HorizonCardHeader(config, data, i18n)

      const html = await TemplateResultTestHelper.renderElement(horizonCardHeader)

      expect(html).toMatchSnapshot()
    })
  })
})
