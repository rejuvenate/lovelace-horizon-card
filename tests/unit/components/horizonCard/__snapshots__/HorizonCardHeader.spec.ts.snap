// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizonCardHeader render does not render the sunrise field when it is not present in the data and it is disabled on the config 1`] = `
<div class="horizon-card-header">
</div>
`;

exports[`HorizonCardHeader render does not render the sunrise field when it is not present in the data but it is activated on the config 1`] = `
<div class="horizon-card-header">
  <div class="horizon-card-text-container">
    <div class="horizon-card-field-name">
      sunrise
    </div>
    <div class="horizon-card-field-value ">
      -
    </div>
  </div>
</div>
`;

exports[`HorizonCardHeader render does not render the sunrise field when it is present in the data but it is disabled on the config 1`] = `
<div class="horizon-card-header">
</div>
`;

exports[`HorizonCardHeader render does not render the sunset field when it is not present in the data and it is disabled on the config 1`] = `
<div class="horizon-card-header">
</div>
`;

exports[`HorizonCardHeader render does not render the sunset field when it is not present in the data but it is activated on the config 1`] = `
<div class="horizon-card-header">
  <div class="horizon-card-text-container">
    <div class="horizon-card-field-name">
      sunset
    </div>
    <div class="horizon-card-field-value ">
      -
    </div>
  </div>
</div>
`;

exports[`HorizonCardHeader render does not render the sunset field when it is present in the data but it is disabled on the config 1`] = `
<div class="horizon-card-header">
</div>
`;

exports[`HorizonCardHeader render does not render the title if it is not present in the configuration 1`] = `
<div class="horizon-card-header">
</div>
`;

exports[`HorizonCardHeader render renders the sunrise field when it is present in the data and it is activated on the config 1`] = `
<div class="horizon-card-header">
  <div class="horizon-card-text-container">
    <div class="horizon-card-field-name">
      sunrise
    </div>
    <div class="horizon-card-field-value ">
      0
    </div>
  </div>
</div>
`;

exports[`HorizonCardHeader render renders the sunset field when it is present in the data and it is activated on the config 1`] = `
<div class="horizon-card-header">
  <div class="horizon-card-text-container">
    <div class="horizon-card-field-name">
      sunset
    </div>
    <div class="horizon-card-field-value ">
      0
    </div>
  </div>
</div>
`;

exports[`HorizonCardHeader render renders the title if it is present in the configuration 1`] = `
<div class="horizon-card-title">
  test
</div>
<div class="horizon-card-header">
</div>
`;
